src/go/configinfo/service_info.go

```diff
@@ -419,7 +422,7 @@ func (s *ServiceInfo) processHttpRule() error {
 	return nil
 }
 
-func (s *ServiceInfo) addOptionMethod(originalMethod *MethodInfo, path string) error {
+func (s *ServiceInfo) addOptionMethod(originalMethod *MethodInfo, httpRule *httppattern.Pattern) error {
 	genOperation := fmt.Sprintf("%s.%s_CORS_%s", originalMethod.ApiName, util.AutogeneratedOperationPrefix, originalMethod.ShortName)
```

Comment:
Maybe add a sanity check to ensure the HTTP method in the httpRule is OPTION. Worried what will happen if its called with a non-option method.

Reply:
Done.